src/Composer/Command/ValidateCommand.php

```diff
@@ -45,6 +45,7 @@ protected function configure()
             ->setDescription('Validates a composer.json and composer.lock.')
             ->setDefinition(array(
                 new InputOption('no-check-all', null, InputOption::VALUE_NONE, 'Do not validate requires for overly strict/loose constraints'),
+                new InputOption('check-lock', null, InputOption::VALUE_NONE, 'Check if lock file is up to date (even config.lock is false, overrides --no-check-lock)'),
```

Comment:
I'd keep this simpler, or leave the flag altogether as I don't really see the point if you disabled the lock file..

```suggestion
                new InputOption('check-lock', null, InputOption::VALUE_NONE, 'Check if lock file is up to date (even when config.lock is false)'),
```